main/src/ca/uwaterloo/flix/language/ast/LiftedAst.scala

```diff
@@ -86,6 +86,12 @@ object LiftedAst {
 
     case class NewObject(name: String, clazz: java.lang.Class[_], tpe: MonoType, purity: Purity, methods: List[JvmMethod], loc: SourceLocation) extends Expr
 
+    case class StructNew(sym: Symbol.StructSym, exps: List[(Symbol.StructFieldSym, Expr)], exp: Expr, tpe: MonoType, purity: Purity, loc: SourceLocation) extends Expr
+
+    case class StructGet(sym: Symbol.StructSym, exp1: Expr, field: Name.Label, tpe: MonoType, purity: Purity, loc: SourceLocation) extends Expr
```

Comment:
You should consider if at least Get and Put could be atomics. That would save you alot of work.